### PR TITLE
settings: make StateMachineSetting.setToDefault a no-op

### PR DESCRIPTION
**Pull request by andreimatei**
I think this method didn't make sense: state machine settings (at least
the cluster version, which is currently the only one) don't have
defaults that they can be reset to. We were already refusing SET CLUSTER
SETTING version = DEFAULT.
Also the implementation of that method was hard to parse. I think the
method was only called at startup by the Values.Init() method, at which
point I think it was a no-op / nonsensical (I think it was setting the
cluster version to 0.0).

Release note: None

### CODE REVIEW

**Review by andreimatei — COMMENTED** [review]
@tbg you said LGTM for this?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34156#-:-LWqTW2uDeYyM8YiO6lC:bl9db6a)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

**Comment by tbg** [comment]
LGTM

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34156#-:-LWr5PzuDbzNLzyxZya2:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34156#-:-LWs6-HjAPYI5-ONx4hw:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)
